### PR TITLE
Fix unsupported distro trigger when no grypedb is present

### PR DESCRIPTION
**Pull request by zburstein**
Changes unsupported distro trigger in grype to use `get_most_recent_active_grypedb` instead of using the grype wrapper to query the record source counts. This resolves a number of issues surfaced by @nightfurys  as well as fixes errors ocurring when grypedb is not fully synced

### CODE REVIEW

File: anchore_engine/services/policy_engine/engine/policy/gate_util_provider.py [hunk 1]
```diff
@@ -119,9 +118,19 @@ def oldest_namespace_feed_sync(
...
+            try:
+                grypedb = get_most_recent_active_grypedb(session)
+            except NoActiveGrypeDB:
+                logger.info(
+                    "No vulnerabilities for image distro found because no active grype db found"
+                )
+                return False
+
+            groups = [
+                group["name"] for group in grypedb.groups if group["record_count"] > 0
```
Comment (nightfurys) [inline anchore_engine/services/policy_engine/engine/policy/gate_util_provider.py:131]: Can grypedb.groups be an invalid reference? Might be worth a check before accessing it

Comment (zburstein) [reply anchore_engine/services/policy_engine/engine/policy/gate_util_provider.py:131]: @nightfurys  Do you mean in the case that grypedb is potentially None? That should be handled by `get_most_recent_active_grypedb` which raises NoActiveGrypeDB if the query does not return any results. I can add that check for safety here as well though, but want to confirm that is what you mean by invalid reference.

Comment (zhill) [reply anchore_engine/services/policy_engine/engine/policy/gate_util_provider.py:131]: 2 cases I can see: grypedb = None, or grypdb.groups = None.

**Review by zhill — APPROVED** [review]
lgtm

File: anchore_engine/services/policy_engine/engine/policy/gate_util_provider.py (hunk 1 above)
Comment (zhill) [reply anchore_engine/services/policy_engine/engine/policy/gate_util_provider.py:131]: I think yes, grypedb=None or grypedb.groups=None. If those are handled in the contract of that function call that should cover it.

Comment (zburstein) [reply anchore_engine/services/policy_engine/engine/policy/gate_util_provider.py:134]: Oh i just pushed the changes to add that right before I saw this response @zhill

Comment (zburstein) [reply anchore_engine/services/policy_engine/engine/policy/gate_util_provider.py:134]: `get_most_recent_active_grypedb` does not check if groups is None so the addition of that check here in this function is a good one

Comment (nightfurys) [reply anchore_engine/services/policy_engine/engine/policy/gate_util_provider.py:134]: @zburstein I meant grypedb.groups != None should be checked. Was that check been added?

**Pull request merged**